src/main/java/org/jabref/gui/entryeditor/SourceTab.java

```diff
@@ -116,11 +116,6 @@ public boolean shouldShow(BibEntry entry) {
 
     @Override
     protected void bindToEntry(BibEntry entry) {
-        // store source if new entry is selected in the maintable and the source tab is focused
-        if (codeArea != null && codeArea.focusedProperty().get()) {
-            DefaultTaskExecutor.runInJavaFXThread(() -> storeSource());
-        }
-
```

Comment:
The problem here is btw that you run it in a separate thread. This means, the moment `storeSource` starts to work, a new entry is already set and what you do is to overwrite the correct new entry with itself.